commit 731a05714e71ace0c9dfa7c0860db758071b740e
Author: Rob Hoes <rob.hoes@citrix.com>
Date:   Tue Mar 13 17:23:39 2012 +0000

    CA-74932: networkd: remove existing IPv4 addresses when configuring static IP
    
    Signed-off-by: Rob Hoes <rob.hoes@citrix.com>

--- a/ocaml/network/network_server.ml
+++ b/ocaml/network/network_server.ml
@@ -221,6 +221,7 @@
 		| Static4 addrs ->
 			if Dhclient.is_running name then
 				ignore (Dhclient.stop name);
+			Ip.flush_ip_addr name;
 			List.iter (Ip.set_ip_addr name) addrs
 
 	let get_ipv4_gateway _ name =
--- a/ocaml/network/network_utils.ml
+++ b/ocaml/network/network_utils.ml
@@ -326,7 +326,9 @@
 
 	let flush_ip_addr ?(ipv6=false) dev =
 		let ipv6' = if ipv6 then ["-6"] else ["-4"] in
-		ignore (call(ipv6' @ ["addr"; "flush"; "dev"; dev]))
+		try
+			ignore (call(ipv6' @ ["addr"; "flush"; "dev"; dev]))
+		with _ -> ()
 
 	let route_show ?(version=V46) dev =
 		let v = string_of_version version in
